setup.py

```diff
@@ -71,6 +71,9 @@ def run(self):
 if sys.version_info < (3, 6):
     extras["dev"].remove("black")
 
+if sys.version_info < (3, 7):
+    del extras["dask_cloudprovider"]
```

Comment:
I don't think we want to remove the extra from users (some users may be relying on this, though at this point 3.6 users are in the minority). Rather, we just want to drop it from our 3.6 test builds for now (though that may be tricky with our ci setup).